test/test_tensor_creation_ops.py

```diff
@@ -1255,6 +1255,14 @@ def test_randperm(self, device):
             torch.randperm(n, out=non_contiguous_tensor)
             self.assertEqual(non_contiguous_tensor, res)
 
+        # Test exceptions when device and generator types are incompatible
+        cuda_gen = torch.Generator(device='cuda')
+        cpu_gen = torch.Generator(device='cpu')
+        for n in (0, 3, 100, 30000):
```

Comment:
Another is: if the slowest part of it is the `n == 30000`, we can probably put larger N values into a SLOW_TEST configuration @samestep has done something very similar in the past

Reply:
yep; for reference: #45785

Reply:
Testing exceptions doesn't take too long even with the big 30000 input, so I think we don't have to move it. Thanks for the reference!